src/System-DependenciesTests/SystemDependenciesTest.class.st

```diff
@@ -593,6 +593,16 @@ SystemDependenciesTest >> testExternalUIDependencies [
 	self assertCollection: dependencies hasSameElements: self knownUIDependencies
 ]
 
+{ #category : #tests }
+SystemDependenciesTest >> testFilesShouldNotDependOnRandom [
+	"In the past Files was depending on Random-Core only to print a random number in a file name.
+	Now, there is no dependency between Files and Random-Core and if possible we would like to keep it this way in order to extract Random-Core from the KernelGroup of the Bootstrap (and maybe of the bootstrap in itself generally)"
+
+	| dependencies |
+	dependencies := self externalDependendiesOf: #( 'Files' ).
+	self deny: (dependencies includes: 'Random-Core')
```

Comment:
This feels weird. If you care about random things not being pulled. Why not just list the dependencies you want instead of blacklisting some that you do not want?

Reply:
That is something that I would like to do later but it requires to check the current dependencies of Files and think about which ones make sense or not and for now I just wanted to focus the extraction of Random-Core from the bootstrap core

Reply:
Some things seems simple, but then I end up spending hours in simple things and not finishing my original fix x_x